### PR TITLE
Use the role name in the db username

### PR DESCRIPTION
**Pull request by briankassouf**

### CODE REVIEW

File: builtin/logical/database/dbplugin/plugin.go [hunk 1]
```diff
@@ -89,7 +96,7 @@ func PluginFactory(pluginName string, sys pluginutil.LookRunnerUtil, logger log.
 // This prevents users from executing bad plugins or executing a plugin
 // directory. It is a UX feature, not a security feature.
 var handshakeConfig = plugin.HandshakeConfig{
-	ProtocolVersion:  1,
+	ProtocolVersion:  2,
```
Comment (briankassouf) [inline builtin/logical/database/dbplugin/plugin.go]: Bumping this means all existing plugins will need to recompile and probably update their code.

File: builtin/logical/database/dbplugin/plugin.go [hunk 2]
```diff
@@ -29,6 +29,13 @@ type Statements struct {
 	RenewStatements      string `json:"renew_statements" mapstructure:"renew_statements" structs:"renew_statements"`
 }
 
+// UsernameConfig is used to configure prefixes for the username to be
+// generated.
+type UsernameConfig struct {
```
Comment (calvn) [inline builtin/logical/database/dbplugin/plugin.go]: What would be a case where `DisplayName` and `RoleName` are different?

Comment (briankassouf) [inline builtin/logical/database/dbplugin/plugin.go]: Display names are tied to tokens- the vault user who created the cred. Role name is the name of the database role that created the cred

Comment (calvn) [inline builtin/logical/database/dbplugin/plugin.go]: I see, so the role name (and unix timestamp) is now included as part of the username for more verbosity as to who and when the token was created?

Comment (briankassouf) [inline builtin/logical/database/dbplugin/plugin.go]: Correct!

**Pull request merged**